### DISCUSSION TITLE
Wrong addComponents padding shorthand property in production?

### DISCUSSION THREAD

**Discussion by gremo** [tailwindlabs/tailwindcss#8535] [Help] [answered]
**What version of Tailwind CSS are you using?**

3.0.24

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss-cli 8.4.13

**What version of Node.js are you using?**

v16.15.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 10 x64

**Reproduction URL**

> **Please note**: this is *not* happening in dev/play, only in prod mode

https://play.tailwindcss.com/uQ38am7QWl

**Describe your issue**

In production, the 4 `padding` properties of my plugin:

```js
addComponents({
  '.vbar': {
    '--vbar-s-factor': '4',

    'padding-top': 'calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-st-factor))',
    'padding-right': 'calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-sr-factor))',
    'padding-bottom': 'calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-sb-factor))',
    'padding-left': 'calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-sl-factor))',
  }
});
```
Becomes:

```
padding: calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-st-factor)) calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-sr-factor)) calc(var(--vbar-s-factor)*var(--vbar-size)*var(--vbar-sb-factor)) calc(var(--vbar-s-factor) * var(--vbar-size) * var(--vbar-sl-factor))
```

![Immagine 2022-06-07 120038](https://user-images.githubusercontent.com/1532616/172353535-a3e8a785-32e8-4dc3-8a91-f535f2b7d9da.png)

... and the padding doesn't work. Can't figure out what's wrong (and why Tailwind is optimizing the padding property itself). Removing one of the 4 properties (i.e. `padding-left`) makes it work again.

Is this a bug or my fault? Any help is much appreciated.

**Comment by adamwathan** [comment, marked as answer]
I pasted your CSS into an empty JSFiddle and it works as expected assuming all of the variables are actually defined:

https://jsfiddle.net/o6xer1Lm/1/

Are any of those variables undefined in your project? In your demo `--vbar-st-factor`, `--vbar-sr-factor`, and `--vbar-sb-factor` are not defined, so if the same is true in your real project that would explain the problem.

What's happening is you are using some CSS minifier and it is combining those rules into a shorthand rule — Tailwind isn't doing that if you are using PostCSS CLI. If you are using cssnano it's this optimization:

https://cssnano.co/docs/optimisations/mergelonghand/

**Reply by gremo** [reply]
Great explanation! I've arrived to the same conclusion after digging into the source code. The problem is exactly... not all variables are defined (because the spacing is not the same). So i'm quite screwed because of the merging feature :)
